string promp_name
int8 input_dofs
float32[] input
